{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib as mpl\n",
    "import numpy as np\n",
    "import matplotlib.patches as mpatches\n",
    "import matplotlib.lines as mlines\n",
    "import matplotlib.transforms as mtransforms\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tools precision/recall/F1score \n",
    "Fig1 SupplementaryFig3 SupplementaryFig4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv='SNV_average.txt'\n",
    "indel='INDEL_average.txt'\n",
    "snv_df=pd.read_csv(snv,sep='\\t')\n",
    "indel_df=pd.read_csv(indel,sep='\\t')\n",
    "shape={'SRR13076390': 'o', 'SRR13076391': 'o', 'SRR13076392': 'o', \n",
    "       'SRR13076393': 'v', 'SRR13076394': 'v','SRR13076395': 'v',\n",
    "       'SRR13076396': 'P', 'SRR13076397': 'P','SRR13076398': 'P'}\n",
    "color={'SRR13076390': 'royalblue', 'SRR13076391': 'crimson', 'SRR13076392': 'goldenrod',\n",
    "       'SRR13076393': 'royalblue', 'SRR13076394': 'crimson','SRR13076395': 'goldenrod',\n",
    "       'SRR13076396': 'royalblue', 'SRR13076397': 'crimson','SRR13076398': 'goldenrod'}\n",
    "# label of lib\n",
    "lab1 = mpatches.Patch(color='royalblue', label='Lib1')\n",
    "lab2 = mpatches.Patch(color='crimson', label='Lib2')\n",
    "lab3 = mpatches.Patch(color='goldenrod', label='Lib3')\n",
    "s1 = mlines.Line2D([], [],marker='o', color='black', markersize=40,linestyle='None', label='Agilent')\n",
    "s2 = mlines.Line2D([], [],marker='v', color='black', markersize=40,linestyle='None', label='IDT')\n",
    "s3 = mlines.Line2D([], [],marker='P', color='black', markersize=40,linestyle='None', label='Roche')\n",
    "\n",
    "def data_process(df, pipeline_now, result):\n",
    "    x = df[ df.Pipeline== pipeline_now ]\n",
    "    return x[result]\n",
    "\n",
    "x = np.arange(len(snv_df.Pipeline))\n",
    "\n",
    "\n",
    "snv_df = snv_df.sort_values(by='F1_avg', ascending=False)\n",
    "pipeline = snv_df.Pipeline.unique().tolist()\n",
    "# data arrange\n",
    "snv_data=[]\n",
    "for pip in pipeline:\n",
    "    snv_recall = data_process(snv_df, pip, 'F1score')\n",
    "    snv_data.append(snv_recall)\n",
    "    \n",
    "plt.boxplot(snv_data, labels=pipeline, showmeans=False , positions=range(18), sym=' ')\n",
    "# boxplot\n",
    "plt.xticks(rotation = 290, fontsize=20)\n",
    "\n",
    "\n",
    "# each point\n",
    "sample_list=['SRR13076390', 'SRR13076391', 'SRR13076392', 'SRR13076393', 'SRR13076394', 'SRR13076395', 'SRR13076396', 'SRR13076397', 'SRR13076398']\n",
    "\n",
    "for sample in sample_list: \n",
    "    df = snv_df[snv_df['Sample'] == sample]\n",
    "    plt.scatter(df.Pipeline, df.F1score, marker=shape[sample], color=color[sample], \n",
    "                label=sample,  linewidth=2.5)\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "\n",
    "plt.yticks(fontsize=8)\n",
    "plt.xticks(fontsize=8)\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "DP result Supplementary Fig2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "shape={'SRR13076390': 'o', 'SRR13076391': 'o', 'SRR13076392': 'o', \n",
    "       'SRR13076393': 'v', 'SRR13076394': 'v','SRR13076395': 'v',\n",
    "       'SRR13076396': 'P', 'SRR13076397': 'P','SRR13076398': 'P'}\n",
    "color={'SRR13076390': 'royalblue', 'SRR13076391': 'crimson', 'SRR13076392': 'goldenrod',\n",
    "       'SRR13076393': 'royalblue', 'SRR13076394': 'crimson','SRR13076395': 'goldenrod',\n",
    "       'SRR13076396': 'royalblue', 'SRR13076397': 'crimson','SRR13076398': 'goldenrod'}\n",
    "sample = ['SRR13076390', 'SRR13076391', 'SRR13076392', 'SRR13076393',\n",
    "       'SRR13076394', 'SRR13076395', 'SRR13076396', 'SRR13076397','SRR13076398']\n",
    "tools = ['Bowtie2', 'BWA', 'DRAGEN-A', 'DRAGMAP','HISAT2']\n",
    "df = pd.read_csv('DP_result.txt', sep = '\\t',header = None)\n",
    "# label of lib\n",
    "lab1 = mpatches.Patch(color='royalblue', label='Lib1')\n",
    "lab2 = mpatches.Patch(color='crimson', label='Lib2')\n",
    "lab3 = mpatches.Patch(color='goldenrod', label='Lib3')\n",
    "\n",
    "# label of scatter\n",
    "s1 = mlines.Line2D([], [],marker='o', color='black',linestyle='None', label='Agilent')\n",
    "s2 = mlines.Line2D([], [],marker='v', color='black',linestyle='None', label='IDT')\n",
    "s3 = mlines.Line2D([], [],marker='P', color='black',linestyle='None', label='Roche')\n",
    "df_data=[]\n",
    "\n",
    "for tool in tools:\n",
    "    df_data.append(df[df[1]==tool][2])\n",
    "\n",
    "\n",
    "\n",
    "#scatter plot\n",
    "for s in sample:\n",
    "    df1 = df[df[0]==s]\n",
    "    c = color[s]\n",
    "    m = shape[s]\n",
    "    plt.scatter(df1[1],df1[2],marker=m, color=c,label=s)\n",
    "\n",
    "plt.boxplot(df_data,labels=tools,sym=' ', positions=[0,1,2,3,4], \n",
    "           medianprops={'lw':0}, meanline=True)\n",
    "\n",
    "\n",
    "# xy label\n",
    "plt.title('')\n",
    "plt.ylabel('Depth')\n",
    "\n",
    "plt.show()\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tools intersection F1score result Fig2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import upsetplot\n",
    "from upsetplot import plot\n",
    "\n",
    "dragen_indel = {'DRAGEN-A_DRAGEN-C':1,'Bowtie2_Mutect2':0, 'Bowtie2_TNscope':0, 'BWA_Mutect2':0, 'BWA_TNscope':0,\n",
    "       'DRAGMAP_Mutect2':0, 'DRAGMAP_TNscope':0, 'HISAT2_Mutect2':0,\n",
    "       'HISAT2_TNscope':0, 'Total_Variants':783, 'TP':534, 'Recall':0.990724, 'Precision':0.681992,\n",
    "       'F1score':0.807867}\n",
    "dragen_snv = {'DRAGEN-A_DRAGEN-C':1,'Bowtie2_Mutect2':0, 'Bowtie2_TNscope':0, 'BWA_Mutect2':0, 'BWA_TNscope':0,\n",
    "       'DRAGMAP_Mutect2':0, 'DRAGMAP_TNscope':0, 'HISAT2_Mutect2':0,\n",
    "       'HISAT2_TNscope':0, 'Total_Variants':42294, 'TP':41253, 'Recall':0.989922, 'Precision':0.975387,\n",
    "       'F1score':0.9826}\n",
    "df = pd.read_csv('INDEL_upsetplot_result.txt', sep='\\t')\n",
    "df = df.sort_values(by='F1score', ascending=False)[0:20]\n",
    "df = df.append(dragen_indel, ignore_index=True).sort_values(by='F1score', ascending=False)\n",
    "\n",
    "df = df.set_index(df.columns[0:8].tolist())\n",
    "upsetplot.plot(df['F1score'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Variant Allele Frequency Fig3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2=pd.read_csv('VAF_dataframe.txt', sep='\\t')\n",
    "df2['D_FN'] = (df2['Dragen_A_Dragen_C_FN']/df2['Known'])*100\n",
    "df2['B_FN'] = (df2['bwa_Mutect2_FN']/df2['Known'])*100\n",
    "df2['M_FN'] = (df2['Dragmap_TNscope_FN']/df2['Known'])*100\n",
    "\n",
    "plt.subplots(4,2,figsize=(20,30))\n",
    "bottom=0\n",
    "# Knownset distribution\n",
    "fig1 = plt.subplot2grid((4,2),(0,0),rowspan=1,colspan=2)\n",
    "fig1.bar(df2['Bin']-0.01,df2['Known'],edgecolor='black',width=0.01, align='edge')\n",
    "fig1.margins(x=0)\n",
    "\n",
    "# Dragen\n",
    "\n",
    "fig2=plt.subplot(423)\n",
    "fig2.bar(df2['Bin']-0.01,100, width=0.01, edgecolor='white',align='edge',label='TP',color='lightgray')\n",
    "fig2.bar(df2['Bin']-0.01, df2['D_FN'], width=0.01, bottom=bottom,\n",
    "         edgecolor='black', align='edge',label='FN',color='crimson')\n",
    "fig2.set_xlim([0,1])\n",
    "fig2.set_ylim([0,100])\n",
    "fig2.legend(loc='upper right', prop={'size': 16})\n",
    "fig2.margins(x=0)\n",
    "\n",
    "\n",
    "fig3 = plt.subplot(424)\n",
    "fig3.bar(df2['Bin']-0.01,df2['Dragen_A_Dragen_C_FP'],align='edge', width=0.01, \n",
    "         label='FP',edgecolor='black', color='royalblue')\n",
    "fig3.set_xlim([0,1])\n",
    "fig3.legend(loc='upper right', prop={'size': 16})\n",
    "fig3.margins(x=0)\n",
    "\n",
    "# Bwa\n",
    "fig4=plt.subplot(425)\n",
    "fig4.bar(df2['Bin']-0.01,100, width=0.01, edgecolor='white',align='edge',label='TP',color='lightgray')\n",
    "fig4.bar(df2['Bin']-0.01, df2['B_FN'], width=0.01, bottom=bottom,\n",
    "         edgecolor='black', align='edge',label='FN',color='crimson')\n",
    "fig4.set_xlim([0,1])\n",
    "fig4.set_ylim([0,100])\n",
    "fig4.legend(loc='upper right', prop={'size': 16})\n",
    "fig4.margins(x=0)\n",
    "\n",
    "\n",
    "fig5 = plt.subplot(426)\n",
    "fig5.bar(df2['Bin']-0.01,df2['bwa_Mutect2_FP'],align='edge', width=0.01, \n",
    "         label='FP',edgecolor='black', color='royalblue')\n",
    "fig5.set_xlim([0,1])\n",
    "fig5.legend(loc='upper right', prop={'size': 16})\n",
    "fig5.margins(x=0)\n",
    "\n",
    "# Dragmap\n",
    "\n",
    "fig6 = plt.subplot(427)\n",
    "fig6.bar(df2['Bin']-0.01,100, width=0.01, edgecolor='white',align='edge',label='TP',color='lightgray')\n",
    "fig6.bar(df2['Bin']-0.01, df2['M_FN'], width=0.01, bottom=bottom,\n",
    "         edgecolor='black', align='edge',label='FN',color='crimson')\n",
    "fig6.set_xlim([0,1])\n",
    "fig6.set_ylim([0,100])\n",
    "fig6.legend(loc='upper right', prop={'size': 16})\n",
    "fig6.margins(x=0)\n",
    "\n",
    "fig7 = plt.subplot(428)\n",
    "fig7.bar(df2['Bin']-0.01,df2['Dragmap_TNscope_FP'],align='edge', width=0.01, \n",
    "         label='FP',edgecolor='black', color='royalblue')\n",
    "fig7.set_xlim([0,1])\n",
    "\n",
    "fig7.legend(loc='upper right', prop={'size': 16})\n",
    "fig7.margins(x=0)\n",
    "\n",
    "\n",
    "# title label\n",
    "fig1.text(0.5,-500,'Known Positive VAF',horizontalalignment='center',fontsize=20)\n",
    "\n",
    "for i in [2,4,6]:\n",
    "    exec(\"fig\"+str(i)+\".text(0.5, -10,'Variant Allele Frequency (VAF)', horizontalalignment='center',fontsize=15)\")\n",
    "    exec(\"fig\"+str(i)+\".text(-0.1,50,'Percentage per bin (%)', verticalalignment='center', fontsize=15,rotation=90)\")\n",
    "    exec(\"fig\"+str(i)+\".set_ylim([0,100])\")\n",
    "\n",
    "for i in [3,5,7]:\n",
    "    exec(\"fig\"+str(i)+\".text(-0.1,25,'Variants Count', verticalalignment='center', fontsize=15,rotation=90)\")\n",
    "    exec(\"fig\"+str(i)+\".text(0.5, -5,'Variant Allel Frequency*', horizontalalignment='center',fontsize=15)\")\n",
    "    exec(\"fig\"+str(i)+\".set_ylim([0,50])\")\n",
    "D = [593, 212, 128, 64]\n",
    "B = [639,323,167,83]\n",
    "M = [238,85,122,69]\n",
    "\n",
    "#\n",
    "f=6\n",
    "fig3.text(0, 51,'593', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig5.text(0, 51,'639', horizontalalignment='left',fontsize=f,rotation=90)\n",
    "fig7.text(0, 51,'238', horizontalalignment='left',fontsize=f,rotation=90)\n",
    "#\n",
    "fig3.text(0.012, 51,'212', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig5.text(0.012, 51,'323', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig7.text(0.012, 51,'85', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "#\n",
    "fig3.text(0.024, 51,'128', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig5.text(0.024, 51,'167', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig7.text(0.024, 51,'122', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "#\n",
    "fig3.text(0.036, 51,'64', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig5.text(0.036, 51,'83', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "fig7.text(0.036, 51,'69', horizontalalignment='left', fontsize=f,rotation=90)\n",
    "\n",
    "for i in [2,3,4,5,6,7]:\n",
    "    exec(\"fig\"+str(i)+\".xaxis.set_tick_params(labelsize=14)\")\n",
    "    exec(\"fig\"+str(i)+\".yaxis.set_tick_params(labelsize=14)\")\n",
    "\n",
    "\n",
    "fig1.text(-0.05,5800, 'A', fontsize=20)\n",
    "fig2.text(-0.1,100,'B',fontsize=20)\n",
    "fig4.text(-0.1,100,'D',fontsize=20)\n",
    "fig6.text(-0.1,100,'F',fontsize=20)\n",
    "\n",
    "fig3.text(-0.1,50,'C',fontsize=20)\n",
    "fig5.text(-0.1,50,'E',fontsize=20)\n",
    "fig7.text(-0.1,50,'G',fontsize=20)\n",
    "\n",
    "#\n",
    "fig2.text(0.5, -15, 'DRAGEN-A_DRAGEN-C', fontsize=15, horizontalalignment='center')\n",
    "fig4.text(0.5, -15, 'BWA_Mutect2', fontsize=15, horizontalalignment='center')\n",
    "fig6.text(0.5, -15, 'DRAGMAP_TNscope', fontsize=15, horizontalalignment='center')\n",
    "fig3.text(0.5, -7.5, 'DRAGEN-A_DRAGEN-C', fontsize=15, horizontalalignment='center')\n",
    "fig5.text(0.5, -7.5, 'BWA_Mutect2', fontsize=15, horizontalalignment='center')\n",
    "fig7.text(0.5, -7.5, 'DRAGMAP_TNscope', fontsize=15, horizontalalignment='center')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cancer related Gene result Fig4\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COSMIC cancer genes less 100 samples \n",
    "\n",
    "df = pd.read_csv('Heatmap_less_100_sample_data.txt', header=None, sep='\\t')\n",
    "df.columns=['Sample', 'Aligner','Caller','Gene','Percentage']\n",
    "df['Index'] = df['Sample']+'_'+df['Aligner']+'_'+df['Caller']\n",
    "order = ['S1_Bowtie2_DeepVariant', 'S1_BWA_DeepVariant', 'S1_DRAGEN-A_DeepVariant', 'S1_DRAGMAP_DeepVariant', 'S2_Bowtie2_DeepVariant', 'S2_BWA_DeepVariant', 'S2_DRAGEN-A_DeepVariant', 'S2_DRAGMAP_DeepVariant', 'S3_Bowtie2_DeepVariant', 'S3_BWA_DeepVariant', 'S3_DRAGEN-A_DeepVariant', 'S3_DRAGMAP_DeepVariant', 'S4_Bowtie2_DeepVariant', 'S4_BWA_DeepVariant', 'S4_DRAGEN-A_DeepVariant', 'S4_DRAGMAP_DeepVariant', 'S5_Bowtie2_DeepVariant', 'S5_BWA_DeepVariant', 'S5_DRAGEN-A_DeepVariant', 'S5_DRAGMAP_DeepVariant', 'S6_Bowtie2_DeepVariant', 'S6_BWA_DeepVariant', 'S6_DRAGEN-A_DeepVariant', 'S6_DRAGMAP_DeepVariant', 'S7_Bowtie2_DeepVariant', 'S7_BWA_DeepVariant', 'S7_DRAGEN-A_DeepVariant', 'S7_DRAGMAP_DeepVariant', 'S8_Bowtie2_DeepVariant', 'S8_BWA_DeepVariant', 'S8_DRAGEN-A_DeepVariant', 'S8_DRAGMAP_DeepVariant', 'S9_Bowtie2_DeepVariant', 'S9_BWA_DeepVariant', 'S9_DRAGEN-A_DeepVariant', 'S9_DRAGMAP_DeepVariant', 'S1_Bowtie2_Mutect2', 'S1_BWA_Mutect2', 'S1_DRAGEN-A_Mutect2', 'S1_DRAGMAP_Mutect2', 'S1_HISAT2_Mutect2', 'S2_Bowtie2_Mutect2', 'S2_BWA_Mutect2', 'S2_DRAGEN-A_Mutect2', 'S2_DRAGMAP_Mutect2', 'S2_HISAT2_Mutect2', 'S3_Bowtie2_Mutect2', 'S3_BWA_Mutect2', 'S3_DRAGEN-A_Mutect2', 'S3_DRAGMAP_Mutect2', 'S3_HISAT2_Mutect2', 'S4_Bowtie2_Mutect2', 'S4_BWA_Mutect2', 'S4_DRAGEN-A_Mutect2', 'S4_DRAGMAP_Mutect2', 'S4_HISAT2_Mutect2', 'S5_Bowtie2_Mutect2', 'S5_BWA_Mutect2', 'S5_DRAGEN-A_Mutect2', 'S5_DRAGMAP_Mutect2', 'S5_HISAT2_Mutect2', 'S6_Bowtie2_Mutect2', 'S6_BWA_Mutect2', 'S6_DRAGEN-A_Mutect2', 'S6_DRAGMAP_Mutect2', 'S6_HISAT2_Mutect2', 'S7_Bowtie2_Mutect2', 'S7_BWA_Mutect2', 'S7_DRAGEN-A_Mutect2', 'S7_DRAGMAP_Mutect2', 'S7_HISAT2_Mutect2', 'S8_Bowtie2_Mutect2', 'S8_BWA_Mutect2', 'S8_DRAGEN-A_Mutect2', 'S8_DRAGMAP_Mutect2', 'S8_HISAT2_Mutect2', 'S9_Bowtie2_Mutect2', 'S9_BWA_Mutect2', 'S9_DRAGEN-A_Mutect2', 'S9_DRAGMAP_Mutect2', 'S9_HISAT2_Mutect2', 'S1_Bowtie2_TNscope', 'S1_BWA_TNscope', 'S1_DRAGEN-A_TNscope', 'S1_DRAGMAP_TNscope', 'S1_HISAT2_TNscope', 'S2_Bowtie2_TNscope', 'S2_BWA_TNscope', 'S2_DRAGEN-A_TNscope', 'S2_DRAGMAP_TNscope', 'S2_HISAT2_TNscope', 'S3_Bowtie2_TNscope', 'S3_BWA_TNscope', 'S3_DRAGEN-A_TNscope', 'S3_DRAGMAP_TNscope', 'S3_HISAT2_TNscope', 'S4_Bowtie2_TNscope', 'S4_BWA_TNscope', 'S4_DRAGEN-A_TNscope', 'S4_DRAGMAP_TNscope', 'S4_HISAT2_TNscope', 'S5_Bowtie2_TNscope', 'S5_BWA_TNscope', 'S5_DRAGEN-A_TNscope', 'S5_DRAGMAP_TNscope', 'S5_HISAT2_TNscope', 'S6_Bowtie2_TNscope', 'S6_BWA_TNscope', 'S6_DRAGEN-A_TNscope', 'S6_DRAGMAP_TNscope', 'S6_HISAT2_TNscope', 'S7_Bowtie2_TNscope', 'S7_BWA_TNscope', 'S7_DRAGEN-A_TNscope', 'S7_DRAGMAP_TNscope', 'S7_HISAT2_TNscope', 'S8_Bowtie2_TNscope', 'S8_BWA_TNscope', 'S8_DRAGEN-A_TNscope', 'S8_DRAGMAP_TNscope', 'S8_HISAT2_TNscope', 'S9_Bowtie2_TNscope', 'S9_BWA_TNscope', 'S9_DRAGEN-A_TNscope', 'S9_DRAGMAP_TNscope', 'S9_HISAT2_TNscope', 'S1_Bowtie2_DRAGEN-C', 'S1_BWA_DRAGEN-C', 'S1_DRAGEN-A_DRAGEN-C', 'S1_DRAGMAP_DRAGEN-C', 'S2_Bowtie2_DRAGEN-C', 'S2_BWA_DRAGEN-C', 'S2_DRAGEN-A_DRAGEN-C', 'S2_DRAGMAP_DRAGEN-C', 'S3_Bowtie2_DRAGEN-C', 'S3_BWA_DRAGEN-C', 'S3_DRAGEN-A_DRAGEN-C', 'S3_DRAGMAP_DRAGEN-C', 'S4_Bowtie2_DRAGEN-C', 'S4_BWA_DRAGEN-C', 'S4_DRAGEN-A_DRAGEN-C', 'S4_DRAGMAP_DRAGEN-C', 'S5_Bowtie2_DRAGEN-C', 'S5_BWA_DRAGEN-C', 'S5_DRAGEN-A_DRAGEN-C', 'S5_DRAGMAP_DRAGEN-C', 'S6_Bowtie2_DRAGEN-C', 'S6_BWA_DRAGEN-C', 'S6_DRAGEN-A_DRAGEN-C', 'S6_DRAGMAP_DRAGEN-C', 'S7_Bowtie2_DRAGEN-C', 'S7_BWA_DRAGEN-C', 'S7_DRAGEN-A_DRAGEN-C', 'S7_DRAGMAP_DRAGEN-C', 'S8_Bowtie2_DRAGEN-C', 'S8_BWA_DRAGEN-C', 'S8_DRAGEN-A_DRAGEN-C', 'S8_DRAGMAP_DRAGEN-C', 'S9_Bowtie2_DRAGEN-C', 'S9_BWA_DRAGEN-C', 'S9_DRAGEN-A_DRAGEN-C', 'S9_DRAGMAP_DRAGEN-C']\n",
    "df1 = df.pivot(index='Gene', columns='Index', values='Percentage')\n",
    "df2 = df1.reset_index()\n",
    "df2 = df2.set_index('Gene')\n",
    "df2=df2[order]\n",
    "\n",
    "\n",
    "i=1\n",
    "select = [e for e in order if \"S1\" in e]\n",
    "\n",
    "exec(\"data = df2[select]\")\n",
    "exec(\"sns.heatmap(data=data, annot=False,cmap='YlGnBu',square=True,cbar=True,cbar_kws={'shrink': .32},\"+\n",
    "     \"linecolor='w',linewidth=0.1)\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drug associate genes result Fig5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Drug associate genes\n",
    "# data transform\n",
    "df = pd.read_csv('Drug_associate_gene_data.txt', header=None, sep='\\t')\n",
    "df.columns=['Sample', 'Aligner','Caller','Gene','Percentage']\n",
    "df['Index'] = df['Sample']+'_'+df['Aligner']+'_'+df['Caller']\n",
    "order = ['S1_Bowtie2_DeepVariant', 'S1_BWA_DeepVariant', 'S1_DRAGEN-A_DeepVariant', 'S1_DRAGMAP_DeepVariant', 'S2_Bowtie2_DeepVariant', 'S2_BWA_DeepVariant', 'S2_DRAGEN-A_DeepVariant', 'S2_DRAGMAP_DeepVariant', 'S3_Bowtie2_DeepVariant', 'S3_BWA_DeepVariant', 'S3_DRAGEN-A_DeepVariant', 'S3_DRAGMAP_DeepVariant', 'S4_Bowtie2_DeepVariant', 'S4_BWA_DeepVariant', 'S4_DRAGEN-A_DeepVariant', 'S4_DRAGMAP_DeepVariant', 'S5_Bowtie2_DeepVariant', 'S5_BWA_DeepVariant', 'S5_DRAGEN-A_DeepVariant', 'S5_DRAGMAP_DeepVariant', 'S6_Bowtie2_DeepVariant', 'S6_BWA_DeepVariant', 'S6_DRAGEN-A_DeepVariant', 'S6_DRAGMAP_DeepVariant', 'S7_Bowtie2_DeepVariant', 'S7_BWA_DeepVariant', 'S7_DRAGEN-A_DeepVariant', 'S7_DRAGMAP_DeepVariant', 'S8_Bowtie2_DeepVariant', 'S8_BWA_DeepVariant', 'S8_DRAGEN-A_DeepVariant', 'S8_DRAGMAP_DeepVariant', 'S9_Bowtie2_DeepVariant', 'S9_BWA_DeepVariant', 'S9_DRAGEN-A_DeepVariant', 'S9_DRAGMAP_DeepVariant', 'S1_Bowtie2_Mutect2', 'S1_BWA_Mutect2', 'S1_DRAGEN-A_Mutect2', 'S1_DRAGMAP_Mutect2', 'S1_HISAT2_Mutect2', 'S2_Bowtie2_Mutect2', 'S2_BWA_Mutect2', 'S2_DRAGEN-A_Mutect2', 'S2_DRAGMAP_Mutect2', 'S2_HISAT2_Mutect2', 'S3_Bowtie2_Mutect2', 'S3_BWA_Mutect2', 'S3_DRAGEN-A_Mutect2', 'S3_DRAGMAP_Mutect2', 'S3_HISAT2_Mutect2', 'S4_Bowtie2_Mutect2', 'S4_BWA_Mutect2', 'S4_DRAGEN-A_Mutect2', 'S4_DRAGMAP_Mutect2', 'S4_HISAT2_Mutect2', 'S5_Bowtie2_Mutect2', 'S5_BWA_Mutect2', 'S5_DRAGEN-A_Mutect2', 'S5_DRAGMAP_Mutect2', 'S5_HISAT2_Mutect2', 'S6_Bowtie2_Mutect2', 'S6_BWA_Mutect2', 'S6_DRAGEN-A_Mutect2', 'S6_DRAGMAP_Mutect2', 'S6_HISAT2_Mutect2', 'S7_Bowtie2_Mutect2', 'S7_BWA_Mutect2', 'S7_DRAGEN-A_Mutect2', 'S7_DRAGMAP_Mutect2', 'S7_HISAT2_Mutect2', 'S8_Bowtie2_Mutect2', 'S8_BWA_Mutect2', 'S8_DRAGEN-A_Mutect2', 'S8_DRAGMAP_Mutect2', 'S8_HISAT2_Mutect2', 'S9_Bowtie2_Mutect2', 'S9_BWA_Mutect2', 'S9_DRAGEN-A_Mutect2', 'S9_DRAGMAP_Mutect2', 'S9_HISAT2_Mutect2', 'S1_Bowtie2_TNscope', 'S1_BWA_TNscope', 'S1_DRAGEN-A_TNscope', 'S1_DRAGMAP_TNscope', 'S1_HISAT2_TNscope', 'S2_Bowtie2_TNscope', 'S2_BWA_TNscope', 'S2_DRAGEN-A_TNscope', 'S2_DRAGMAP_TNscope', 'S2_HISAT2_TNscope', 'S3_Bowtie2_TNscope', 'S3_BWA_TNscope', 'S3_DRAGEN-A_TNscope', 'S3_DRAGMAP_TNscope', 'S3_HISAT2_TNscope', 'S4_Bowtie2_TNscope', 'S4_BWA_TNscope', 'S4_DRAGEN-A_TNscope', 'S4_DRAGMAP_TNscope', 'S4_HISAT2_TNscope', 'S5_Bowtie2_TNscope', 'S5_BWA_TNscope', 'S5_DRAGEN-A_TNscope', 'S5_DRAGMAP_TNscope', 'S5_HISAT2_TNscope', 'S6_Bowtie2_TNscope', 'S6_BWA_TNscope', 'S6_DRAGEN-A_TNscope', 'S6_DRAGMAP_TNscope', 'S6_HISAT2_TNscope', 'S7_Bowtie2_TNscope', 'S7_BWA_TNscope', 'S7_DRAGEN-A_TNscope', 'S7_DRAGMAP_TNscope', 'S7_HISAT2_TNscope', 'S8_Bowtie2_TNscope', 'S8_BWA_TNscope', 'S8_DRAGEN-A_TNscope', 'S8_DRAGMAP_TNscope', 'S8_HISAT2_TNscope', 'S9_Bowtie2_TNscope', 'S9_BWA_TNscope', 'S9_DRAGEN-A_TNscope', 'S9_DRAGMAP_TNscope', 'S9_HISAT2_TNscope', 'S1_Bowtie2_DRAGEN-C', 'S1_BWA_DRAGEN-C', 'S1_DRAGEN-A_DRAGEN-C', 'S1_DRAGMAP_DRAGEN-C', 'S2_Bowtie2_DRAGEN-C', 'S2_BWA_DRAGEN-C', 'S2_DRAGEN-A_DRAGEN-C', 'S2_DRAGMAP_DRAGEN-C', 'S3_Bowtie2_DRAGEN-C', 'S3_BWA_DRAGEN-C', 'S3_DRAGEN-A_DRAGEN-C', 'S3_DRAGMAP_DRAGEN-C', 'S4_Bowtie2_DRAGEN-C', 'S4_BWA_DRAGEN-C', 'S4_DRAGEN-A_DRAGEN-C', 'S4_DRAGMAP_DRAGEN-C', 'S5_Bowtie2_DRAGEN-C', 'S5_BWA_DRAGEN-C', 'S5_DRAGEN-A_DRAGEN-C', 'S5_DRAGMAP_DRAGEN-C', 'S6_Bowtie2_DRAGEN-C', 'S6_BWA_DRAGEN-C', 'S6_DRAGEN-A_DRAGEN-C', 'S6_DRAGMAP_DRAGEN-C', 'S7_Bowtie2_DRAGEN-C', 'S7_BWA_DRAGEN-C', 'S7_DRAGEN-A_DRAGEN-C', 'S7_DRAGMAP_DRAGEN-C', 'S8_Bowtie2_DRAGEN-C', 'S8_BWA_DRAGEN-C', 'S8_DRAGEN-A_DRAGEN-C', 'S8_DRAGMAP_DRAGEN-C', 'S9_Bowtie2_DRAGEN-C', 'S9_BWA_DRAGEN-C', 'S9_DRAGEN-A_DRAGEN-C', 'S9_DRAGMAP_DRAGEN-C']\n",
    "df1 = df.pivot(index='Gene', columns='Index', values='Percentage')\n",
    "df2 = df1.reset_index()\n",
    "df2 = df2.set_index('Gene')\n",
    "df2=df2[order]\n",
    "\n",
    "\n",
    "\n",
    "i=1\n",
    "select = [e for e in order if \"S1\" in e]\n",
    "exec(\"data = df2[select]\")\n",
    "exec(\"sns.heatmap(data=data, annot=False,cmap='YlGnBu',square=True,cbar=True,cbar_kws={'shrink': .32},\"+\n",
    "        \"linecolor='w',linewidth=0.1)\")\n",
    "exec(\"fig\"+str(i)+\".set(xlabel='', ylabel='')\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tumor mutation burden Supplementary Fig9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('TMB.txt', sep='\\t')\n",
    "shape={'SRR13076390': 'o', 'SRR13076391': 'o', 'SRR13076392': 'o', \n",
    "       'SRR13076393': 'v', 'SRR13076394': 'v','SRR13076395': 'v',\n",
    "       'SRR13076396': 'P', 'SRR13076397': 'P','SRR13076398': 'P'}\n",
    "color={'SRR13076390': 'royalblue', 'SRR13076391': 'crimson', 'SRR13076392': 'goldenrod',\n",
    "       'SRR13076393': 'royalblue', 'SRR13076394': 'crimson','SRR13076395': 'goldenrod',\n",
    "       'SRR13076396': 'royalblue', 'SRR13076397': 'crimson','SRR13076398': 'goldenrod'}\n",
    "sample = ['SRR13076390', 'SRR13076391', 'SRR13076392', 'SRR13076393',\n",
    "       'SRR13076394', 'SRR13076395', 'SRR13076396', 'SRR13076397','SRR13076398']\n",
    "tools_list = ['Bowtie2_Mutect2','DRAGEN-A_Mutect2','DRAGMAP_Mutect2', 'HISAT2_Mutect2',\n",
    " 'BWA_Mutect2','Bowtie2_DRAGEN-C',   'DRAGEN-A_DRAGEN-C','DRAGMAP_DRAGEN-C','BWA_DRAGEN-C', \n",
    " 'Bowtie2_TNscope','DRAGEN-A_TNscope','DRAGMAP_TNscope','HISAT2_TNscope','BWA_TNscope',   \n",
    " 'Bowtie2_DeepVariant','DRAGEN-A_DeepVariant','DRAGMAP_DeepVariant','BWA_DeepVariant']\n",
    "\n",
    "# boxplot\n",
    "df_data = []\n",
    "for tool in tools_list:\n",
    "    df_data.append(df[df['Tools']==tool]['TMB/MB'])\n",
    "\n",
    "plt.boxplot(df_data,labels=tools_list,sym=' ', showmeans=False,\n",
    "           medianprops={'lw':0}, meanline=False, positions=[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17])\n",
    "\n",
    "\n",
    "\n",
    "# scatter\n",
    "for s in sample:\n",
    "    df1 = df[df['Sample'] == s]\n",
    "    mark = shape[s]\n",
    "    c = color[s]\n",
    "    plt.scatter(df1['position'],df1['TMB/MB'], marker=mark, color=c, label=s)\n",
    "\n",
    "plt.xticks(rotation=90)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Supplementary Fig5-8 are drawn by R "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "display_venn <- function(input){\n",
    "  library(VennDiagram)\n",
    "  df = read.delim(input,sep='\\t',header=FALSE)\n",
    "  x<-list(bowtie2=df$V1,bwa=df$V2,dragen=df$V3,dragmap=df$V4,hisat2=df$V5)\n",
    "  color_list = c('#f5c500','#fc88b3','#a8a4eb','#8ed6a4','#6bccf4')\n",
    "  grid.newpage()\n",
    "  pushViewport(viewport(width=unit(0.9, \"npc\"), height = unit(0.9, \"npc\")))\n",
    "  venn_object <- venn.diagram(x,category.names=c('bowtie2', 'bwa', 'Dragen', 'Dragmap', 'Hisat2'),\n",
    "                              filename=NULL,\n",
    "                              # circle\n",
    "                              lty = 0,\n",
    "                              lwd = 2,\n",
    "                              col = color_list,\n",
    "                              alpha=c(0.35,0.35,0.35,0.35,0.35),\n",
    "                              fill=color_list,\n",
    "                              cat.col=color_list,\n",
    "                              # fontsize\n",
    "                              cat.cex=1.5,\n",
    "                              cex=1.5)\n",
    "  grid.draw(venn_object)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Supplementary Fig10-15 are drawn by SigProfilerExtractor v1.1.4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# python3.9  sigprofile.py -i ${input} "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
